pkg/gpu/probe.go

```diff
@@ -56,134 +89,80 @@ type Probe struct {
 	procMon        *monitor.ProcessMonitor
 }
 
-// NewProbe starts the GPU monitoring probe, setting up the eBPF program and the uprobes, the
+// NewProbe creates a GPU monitoring probe, containing relevant eBPF programs (uprobes), the
```

Comment:
It's also starting the probe, I'd leave that part in the comment.

```suggestion
// NewProbe creates and starts a GPU monitoring probe, containing relevant eBPF programs (uprobes), the
```